app/src/main/java/com/pajato/argus/ListAdapter.kt

```diff
@@ -14,20 +15,38 @@ class ListAdapter(val items: MutableList<Video>) : RecyclerView.Adapter<ListAdap
     }
 
     override fun onBindViewHolder(holder: ViewHolder, position: Int) {
-        val titleTextView: TextView = holder.layout.findViewWithTag("title")
+        // Fill in the data and set an onClick for the delete button
+        val titleTextView = holder.layout.findViewWithTag<TextView>("title")
         titleTextView.text = items[position].title
-        val networkTextView: TextView = holder.layout.findViewWithTag("network")
+        val networkTextView = holder.layout.findViewWithTag<TextView>("network")
         networkTextView.text = items[position].network
+        val deleteButton = holder.layout.findViewWithTag<AppCompatImageButton>("delete")
+        deleteButton.setOnClickListener(Delete(holder, this))
     }
 
     override fun getItemCount(): Int {
         return items.size
     }
 
-    fun addItem(video : Video) {
+    fun addItem(video: Video) {
         items.add(video)
         this.notifyItemInserted(items.size - 1)
     }
 
+    fun removeItem(position : Int) {
+        items.removeAt(position)
+        notifyItemRemoved(position)
+        notifyItemRangeChanged(position, items.size)
+    }
+
+    // The ViewHolder class
     class ViewHolder(val layout: View) : RecyclerView.ViewHolder(layout)
+
+    // An onClick that deletes the item from the adapter and removes it from the database.
+    class Delete(private val holder: ViewHolder, private val adapter: ListAdapter) : View.OnClickListener {
+        override fun onClick(v: View?) {
+            DatabaseHelper.deleteVideo(adapter.items[holder.adapterPosition], holder.layout.context)
```

Comment:
There is no way that the list adapter class wants to know about a database in the app.  Period. Rather the list adapter class wants to listen for requests to delete/remove (or add) a list item.  The UI wants to post requests to add/remove/delete list items. In this way the classes are loosely coupled which provides for easily maintained code.  This is a very important and good thing.